### ISSUE TITLE
Binding to the 'latest' version of a secret should match k8s secrets behavior

### ISSUE THREAD

**Issue by tam7t** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#14]
### TL;DR

When a K8S `Secret` is mounted into a pod filesystem, and then mutated in the K8S API, the value in the pod filesystem is updated.

This behavior should be possible when using the `secrets-store-csi-driver-provider-gcp` plugin within some time bound (~5 minutes).

### Design

**Proposal**

The driver currently only executes and fetches secrets on the `mount` event.  When the `SecretProviderClass` creates a binding to a 'latest' version of a secret:

```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: app-secrets
spec:
  provider: gcp
  parameters:
    secrets: |
      - resourceName: "projects/$PROJECT_ID/secrets/testsecret/versions/latest"
        fileName: "good1.txt"
```

The driver should poll the `secretmanager` API frequently and update the value in the mounted pod filesystem if the value changes, until the pod is deleted.

**Alternatives considered**

**Resources**

- Related issue: https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/125

**Additional information**

**Comment by tam7t** [comment]
This is addressed by https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/docs/README.rotation.md

The current release (v0.2.0) supports this, but it must be enabled manually when installing `secrets-store-csi-driver` with the command line flag `--enable-secret-rotation=false`
